### PR TITLE
MRG, BUG: Fix reading KIT data with fewer HPI coils

### PR DESCRIPTION
**Pull request by larsoner**
@Kyung-minAn can you try this with your data? It fixes the problem for me with `mne.io.read_raw_fif('010409_Motor_task_coregist-export_tiny_1s.con')`

### CODE REVIEW

**Comment by larsoner** [comment]
On 9c93805 just `test_calculate_chpi_snr` should fail. Our testing infrastructure was a bit broken in that we had missed a bunch of `requires_testing_data` decorators. I've refactored it so that tests are now run before any caching/downloading occurs, and left one expected failure in there to test. Assuming that's the only failure, I'll push a commit to decorate `test_calculate_chpi_snr` properly, then all should be green. We'll see!

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -135,6 +135,8 @@ Bugs
 
 - Fix bug with `mne.io.read_raw_fieldtrip` and `mne.read_epochs_fieldtrip` where channel positions were not set properly (:gh:`9447` by `Eric Larson`_)
 
+- Fix bug with :func:`mne.io.read_raw_kit` where omitting HPI coils could lead to an AssertionError on reading (:gh:`9612` by `Eric Larson`_)
```
Comment (hoechenberger) [inline doc/changes/latest.inc:138]:
```suggestion
- Fix bug with :func:`mne.io.read_raw_kit` where omitting HPI coils could lead to an ``AssertionError`` on reading (:gh:`9612` by `Eric Larson`_)
```

**Review by hoechenberger — COMMENTED** [review]
Otherwise LGTM

**Review by hoechenberger — APPROVED** [review]

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx @larsoner !

**Comment by Kyung-minAn** [comment]
Dear @larsoner (cc. @agramfort )

Thank you for your response.

I tried the function 'mne.io.read_raw_fif'. 
There was an error like below.

```
raw = mne.io.read_raw_fif('010409_Motor_task_coregist-export_tiny_1s.con')
Opening raw data file 010409_Motor_task_coregist-export_tiny_1s.con...
C:\Users\eiren\AppData\Local\Temp/ipykernel_30696/3071446323.py:1: RuntimeWarning: This filename (010409_Motor_task_coregist-export_tiny_1s.con) does not conform to MNE naming conventions. All raw files should end with raw.fif, raw_sss.fif, raw_tsss.fif, _meg.fif, _eeg.fif, _ieeg.fif, raw.fif.gz, raw_sss.fif.gz, raw_tsss.fif.gz, _meg.fif.gz, _eeg.fif.gz or _ieeg.fif.gz
  raw = mne.io.read_raw_fif('010409_Motor_task_coregist-export_tiny_1s.con')
Traceback (most recent call last):

  File "C:\Users\eiren\AppData\Local\Temp/ipykernel_30696/3071446323.py", line 1, in <module>
    raw = mne.io.read_raw_fif('010409_Motor_task_coregist-export_tiny_1s.con')

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\io\fiff\raw.py", line 473, in read_raw_fif
    return Raw(fname=fname, allow_maxshield=allow_maxshield,

  File "<decorator-gen-210>", line 24, in __init__

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\io\fiff\raw.py", line 82, in __init__
    self._read_raw_file(next_fname, allow_maxshield,

  File "<decorator-gen-211>", line 24, in _read_raw_file

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\io\fiff\raw.py", line 157, in _read_raw_file
    ff, tree, _ = fiff_open(fname, preload=whole_file)

  File "<decorator-gen-6>", line 24, in fiff_open

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\io\open.py", line 126, in fiff_open
    return _fiff_open(fname, fid, preload)

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\io\open.py", line 144, in _fiff_open
    raise ValueError('file does not start with a file id tag')

ValueError: file does not start with a file id tag
```

Could you check it?

Many thanks,
Min

**Comment by larsoner** [comment]
@Kyung-minAn I think there is a typo/error, you should use `mne.io.read_raw_kit` for KIT data, not `mne.io.read_raw_fif` (which is only for FIF)

**Comment by Kyung-minAn** [comment]
Thank you. @larsoner 

I tried 'mne.io.read_raw_kit' to import the coregistrated-con data.
I got this error massage.

```
raw = mne.io.read_raw_kit('010409_Motor_task_coregist-export_tiny_1s.con')
Extracting SQD Parameters from 010409_Motor_task_coregist-export_tiny_1s.con...
Creating Raw.info structure...
Traceback (most recent call last):

  File "C:\Users\eiren\AppData\Local\Temp/ipykernel_30696/1798575027.py", line 1, in <module>
    raw = mne.io.read_raw_kit('010409_Motor_task_coregist-export_tiny_1s.con')

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\io\kit\kit.py", line 908, in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,

  File "<decorator-gen-234>", line 24, in __init__

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\io\kit\kit.py", line 133, in __init__
    info, kit_info = get_kit_info(

  File "<decorator-gen-237>", line 24, in get_kit_info

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\io\kit\kit.py", line 749, in get_kit_info
    assert meg_done

AssertionError
```

**Comment by larsoner** [comment]
@Kyung-minAn the fixes are in latest `main` as of yesterday, you need to update to that version (e.g., with `pip install --upgrade https://github.com/mne-tools/mne-python/zipball/main`). Line 749 on `main` does not say `assert meg_done` anymore, so you must be on an older version:

https://github.com/mne-tools/mne-python/blob/474da964415c5233e955663b56e85531093be0a4/mne/io/kit/kit.py#L749

**Comment by Kyung-minAn** [comment]
@larsoner Thank you, It is working!! =)

I would like to ask about one further error.
I cannot use the code of 'raw.plot_sensors()' anymore.

```
raw.plot_sensors()
Traceback (most recent call last):

  File "C:\Users\eiren\AppData\Local\Temp/ipykernel_30696/2660701588.py", line 1, in <module>
    raw.plot_sensors()

  File "<decorator-gen-38>", line 24, in plot_sensors

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\channels\channels.py", line 593, in plot_sensors
    return plot_sensors(self.info, kind=kind, ch_type=ch_type, title=title,

  File "<decorator-gen-99>", line 24, in plot_sensors

  File "C:\Users\eiren\anaconda3\envs\mne\lib\site-packages\mne\viz\utils.py", line 887, in plot_sensors
    raise TypeError('info must be an instance of Info not %s' % type(info))

TypeError: info must be an instance of Info not <class 'mne.io.meas_info.Info'>
```

**Comment by larsoner** [comment]
Hmm, that's weird. I cannot replicate with:
```
>>> import mne
>>> raw = mne.io.read_raw_kit('010409_Motor_task_coregist-export_tiny_1s.con')  # now in the testing dataset
>>> raw.plot_sensors()
<Figure size 640x640 with 1 Axes>
```
Maybe you need to restart your kernel if you're using Spyder?

**Comment by Kyung-minAn** [comment]
@larsoner 
As you commented, I restarted my kernel.
It is working well now.
Thank you for fixing the problem and checking it!

I hope you have a wonderful summer!

Many thanks,
Min

**Pull request merged**